src/lazy/text/buffer.rs

```diff
@@ -334,15 +335,31 @@ impl<'data> TextBufferView<'data> {
     pub fn match_sexp_value_1_1(
         self,
     ) -> IonParseResult<'data, Option<LazyRawValueExpr<'data, TextEncoding_1_1>>> {
-        alt((
-            whitespace_and_then(
-                Self::match_e_expression
-                    .map(|matched| Some(RawValueExpr::MacroInvocation(matched))),
-            ),
-            Self::match_sexp_value.map(|maybe_matched| {
+        whitespace_and_then(alt((
+            Self::match_e_expression.map(|matched| Some(RawValueExpr::MacroInvocation(matched))),
+            value(None, tag(")")),
+            pair(
+                opt(Self::match_annotations),
+                // We need the s-expression parser to recognize the input `--3` as the operator `--` and the
```

Comment:
🗺️ Because this delegated to `match_sexp_value` (implied `_1_0`), it would use the Ion 1.0 matchers for matching anything that is not an E-Expression inside a S-Expression, which caused this to happen: 
```
outer::(     // Handled by match_sexp_1_1
  abc        // Handled by match_value_1_0
  (:foo)     // Handled by match_e_expression
  inner::(   // Handled by match_sexp_1_0
    def      // Handled by match_value_1_0
    (:bar)   // Invalid (for match_sexp_1_0)
  )
)
```
The solution was to copy/paste the content from `match_sexp_value`, changing it to use `match_value_1_1` instead of `match_value_1_0`.